ttn/application.py

```diff
@@ -0,0 +1,272 @@
+        if hasattr(self, "app_access_token"):
+            return [("token", self.app_access_token)]
+        elif hasattr(self, "app_access_key"):
+            return [("key", self.app_access_key)]
+
+    def get(self):
+        req = proto.ApplicationIdentifier()
+        req.app_id = self.app_id
+        meta = self.__create_metadata()
+        try:
+            app = self.client.GetApplication(req, 60, meta)
+            return app
+        except grpc.RpcError as err:
+            raise RuntimeError(
+                "Error while getting the",
+                " application: {}".format(err.code().name))
+
+    def set_payload_format(self, payloadf):
+        self.__set({
+          "payload_format": payloadf,
+        })
+
+    def set_custom_payload_functions(self, **kwargs):
+        updates = {
+            "payload_format": "custom",
+        }
+        for k, v in kwargs.items():
+            print(k, v)
+            updates[k] = v
+        self.__set(updates)
+
+    def set_register_on_join_access_key(self, to):
+        self.__set({
+            "register_on_join_access_key": to,
+        })
+
+    def unregister(self):
+        req = proto.ApplicationIdentifier()
+        req.app_id = self.app_id
+        meta = self.__create_metadata()
+        try:
+            return self.client.DeleteApplication(req, 60, meta)
+        except grpc.RpcError as err:
+            raise RuntimeError(
+                "Error while deleting the",
+                " application: {}".format(err.code().name))
+
+    def __set(self, updates):
+        req = proto.Application()
+        req.app_id = self.app_id
+
+        if "payload_format" in updates:
+            req.payload_format = updates["payload_format"]
+
+        if "register_on_join_access_key" in updates:
+            req.register_on_join_access_key = updates[("register_on_"
+                                                       "join_access_key")]
+
+        if "decoder" in updates:
+            req.decoder = updates["decoder"]
+
+        if "converter" in updates:
+            req.converter = updates["converter"]
+
+        if "encoder" in updates:
+            req.encoder = updates["encoder"]
+
+        if "validator" in updates:
+            req.validator = updates["validator"]
+
+        meta = self.__create_metadata()
+        try:
+            return self.client.SetApplication(req, 60, meta)
+        except grpc.RpcError as err:
+            raise RuntimeError(
+                "Error while setting the",
+                " application: {}".format(err.code().name))
+
+    def register_device(self, devID, device):
+        return self.__setDevice(devID, device)
+
+    def __setDevice(self, devID, device):
+        req = self.__deviceRequest(devID, device)
+        meta = self.__create_metadata()
+        try:
+            return self.client.SetDevice(req, 60, meta)
+        except grpc.RpcError as err:
+            raise RuntimeError(
+                "Error while setting the",
```

Comment:
`SetDevice` is an API term, but use another formulation for humans (e.g. *Error when updating the device*)